metrics/ci_integrations/test/destination/firestore/config/validator/firestore_destination_validator_test.dart

```diff
@@ -0,0 +1,586 @@
+    });
+
+    test(
+      "throws an ArgumentError if the given validation delegate is null",
+      () {
+        expect(
+          () => FirestoreDestinationValidator(null, validationResultBuilder),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "throws an ArgumentError if the given validation result builder is null",
+      () {
+        expect(
+          () => FirestoreDestinationValidator(validationDelegate, null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "creates a new instance with the given parameters",
+      () {
+        final validator = FirestoreDestinationValidator(
+          validationDelegate,
+          validationResultBuilder,
+        );
+
+        expect(validator.validationDelegate, equals(validationDelegate));
+        expect(
+          validator.validationResultBuilder,
+          equals(validationResultBuilder),
+        );
+      },
+    );
+
+    test(
+      ".validate() delegates the public api key validation to the validation delegate",
+      () async {
+        whenValidateFirebasePublicApiKey().thenAnswer(
+          (_) => Future.value(failureResult),
+        );
+
+        await validator.validate(config);
+
+        verify(
+          validationDelegate.validatePublicApiKey(firebaseApiKey),
+        ).called(once);
+      },
+    );
+
+    test(
+      ".validate() sets the public api key validation result returned by the validation delegate",
+      () async {
+        whenValidateFirebasePublicApiKey().thenAnswer(
+          (_) => Future.value(failureResult),
+        );
+
+        await validator.validate(config);
+
+        verify(validationResultBuilder.setResult(
+          FirestoreDestinationConfigField.firebasePublicApiKey,
+          failureResult,
+        )).called(once);
+      },
+    );
+
+    test(
+      ".validate() sets empty result with the unknown field validation result with the 'firebase public api key invalid' additional context if the public api key validation fails",
+      () async {
+        const expectedResult = FieldValidationResult.unknown(
+          additionalContext:
+              FirestoreStrings.publicApiKeyInvalidInterruptReason,
+        );
+        whenValidateFirebasePublicApiKey().thenAnswer(
+          (_) => Future.value(failureResult),
+        );
+
+        await validator.validate(config);
+
+        verify(
+          validationResultBuilder.setEmptyResults(expectedResult),
+        ).called(once);
+      },
+    );
+
+    test(
+      ".validate() does not validate the auth if the public api key validation fails",
+      () async {
+        whenValidateFirebasePublicApiKey().thenAnswer(
+          (_) => Future.value(failureResult),
+        );
+
+        await validator.validate(config);
+
+        verifyNever(validationDelegate.validateAuth(any));
+      },
+    );
+
+    test(
+      ".validate() does not validate the firebase project id if the public api key validation fails",
```

Comment:
```suggestion
      ".validate() does not validate the firebase project id if the public api key validation result is failed",
```
Would it be a bit more clear?